crates/http-source/src/formatter/mod.rs

```diff
@@ -0,0 +1,11 @@
+mod http_json_record;
+mod http_response_record;
+mod json_formatter;
+mod response_formatter;
+mod stream_formatter;
+mod text_formatter;
+
+use json_formatter::JsonFormatter;
+pub(crate) use response_formatter::ResponseFormatter;
+pub(crate) use stream_formatter::StreamFormatter;
+use text_formatter::TextFormatter;
```

Comment:
Currently, we have `Formatter` trait and two implementations `JsonFormatter` and `TextFormatter`.

In this PR, you have structs: `JsonFormatter`, `TextFormatter`, `StreamFormatter`, and an enum `ResponseFormatter` that holds the first two. The relationships are unclear. There is no common trait. Common suffixes in names are confusing.